Source/drlg_l1.cpp

```diff
@@ -1673,18 +1696,21 @@ void __fastcall DRLG_L5GHall(int x1, int y1, int x2, int y2)
 
 void __fastcall DRLG_L5SetRoom(int rx1, int ry1)
 {
-	int rw = *(unsigned char *)pSetPiece; /* todo: BYTE */
-	int rh = *((unsigned char *)pSetPiece + 2);
+	int rw, rh, i, j;
+	unsigned char *sp;
+
+	rw = *(unsigned char *)pSetPiece; /* todo: BYTE */
+	rh = *((unsigned char *)pSetPiece + 2);
 
 	setpc_x = rx1;
 	setpc_y = ry1;
 	setpc_w = rw;
 	setpc_h = rh;
 
-	unsigned char *sp = (unsigned char *)pSetPiece + 4;
+	sp = (unsigned char *)pSetPiece + 4;

```

Comment:
Perhaps sp is also `unsigned short *`, that would make the `sp += 2;` into `sp++`.

However, perhaps we change this in the PR that makes these functions bin perfect (if they are not already). So we don't mess them up.

So, upon further thought, perhaps lets focus this PR on C++ to C conversion.

Reply:
Agree, I tried to keep away from any other clean ups as well while doing it.

Reply:
The function is already bin exact so I made an issue for this: https://github.com/diasurgical/devilution/issues/520